feat: 'binary' argument to use a preinstalled cli executable

matt-codecov opened the pull request.

joseph-sentry commented on src/index.ts:
```diff
@@ -64,37 +87,27 @@ try {
+try {
+  const failCi = isTrue(core.getInput('fail_ci_if_error'));
+  const binaryPath = core.getInput('binary');
+  const verbose = isTrue(core.getInput('verbose'));
```
nit: just personal preference so feel free to ignore this one, I think I'd rather get the value for `verbose` in the functions where its needed rather than get it at the top of main and pass it through multiple layers of function calls to get to where its being used

joseph-sentry reviewed (approved):
generally change lgtm, left a couple points about style really, i do think there's a pre-commit hook that rebuilds the app and lints, similar to the coverage action, i think running `install.sh` should install the hook but i'm pretty sure you can also manually run the `pre-commit` script then commit the changes as well

matt-codecov commented on src/index.ts:
```diff
@@ -21,23 +21,46 @@ import versionInfo from './version';
+    );
+  };
+
+  const runCommands = async () => {
```
i just copied what was already there :P

matt-codecov commented:
ty for pointing me at `install.sh`!

PR has been updated. i added test coverage for more things and added some jest boilerplate to manage env vars better while i was at it
